baremetal/src/Messenger.cc

```diff
@@ -10,103 +10,109 @@
 #include <ebbrt/UniqueIOBuf.h>
 
 uint16_t ebbrt::Messenger::port_;
+double ebbrt::Messenger::Connection::occupancy_ratio_ = 0.2;
+uint8_t ebbrt::Messenger::Connection::preallocate_chain_len_ = 100;
 
 ebbrt::Messenger::Messenger() {}
 
 ebbrt::Messenger::Connection::Connection(ebbrt::NetworkManager::TcpPcb pcb)
     : TcpHandler(std::move(pcb)) {}
 
+void ebbrt::Messenger::Connection::preallocated(std::unique_ptr<MutIOBuf> b) {
+
+  auto len = b->Length();
+  buf_->Advance(preallocate_);
```

Comment:
probably a bit clearer to change this to:

    auto ptr = buf_->MutData() + preallocate_;
    std::memcpy(ptr, b->Data(), len);

Not sure if the cast is necessary.